pkg/k8s/ipresolver.go

```diff
@@ -700,22 +701,33 @@ func (resolver *K8sIPResolver) getControllerOfOwner(snapshot *clusterSnapshot, o
 }
 
 func (resolver *K8sIPResolver) resolvePodDescriptor(pod *v1.Pod) Workload {
+	existing, ok := resolver.snapshot.PodDescriptors.Load(pod.UID)
+	if ok {
+		result, ok := existing.(Workload)
+		if ok {
+			return result
+		}
+	}
+	var err error
 	name := pod.Name
 	namespace := pod.Namespace
 	kind := "pod"
 	owner := metav1.GetControllerOf(pod)
 	for owner != nil {
-		var err error
 		name = owner.Name
 		kind = owner.Kind
 		owner, err = resolver.getControllerOfOwner(&resolver.snapshot, owner)
 		if err != nil {
 			log.Printf("Error retreiving owner of %v - %v", name, err)
 		}
 	}
-	return Workload{
+	result := Workload{
 		Name:      name,
 		Namespace: namespace,
 		Kind:      kind,
 	}
+	if err == nil {
```

Comment:
Can we move this after the getControllerOfOwner call? Instead of keeping the error and checking here?

Reply:
Discussed and commented